src/lib.rs

```diff
@@ -272,6 +280,12 @@ macro_rules! ne_vec {
     ($($x:expr),+ $(,)?) => {
         unsafe { $crate::NonEmpty::new_unchecked(vec![$($x),+]) }
     };
+    ($elem:expr; $n:expr) => {{
+        if $n == 0 {
+            ::std::panic!("`NonEmpty` vector must be non-empty");
```

Comment:
is it possible to check this in compile time? not familiar with macros yet.

Reply:
`$n` can be any expression as long as it evaluates to a usize and is not necessarily known at compile time.
Although perhaps it's possible to add another macro where `$n: literal`, so it could at least *try* to assert this at compile time if the given `$n` is a literal (which is likely how it's used most of the time), and fall back to the existing macro if $n is an expression and can only be determined at runtime?
It's possible to do this with the old `[(); N - 1]` hack where compilation fails if N is 0, but unfortunately the error isn't too great (though the error does point out that it's coming from the ne_vec! macro invocation and the constant name suggests that this is due to `$n` being zero):
```rs
macro_rules! ne_vec {
    // in addition to the old macro
    ($elem:expr; $n:literal) => {{
        // if $n is 0, then the next line will cause an error as 0 - 1 underflows
        const _ASSERT_NON_ZERO: [(); $n - 1] = [(); $n - 1];
        unsafe { $crate::NonEmpty::new_unchecked(vec![$elem; $n]) }
    }};
}
```

```rs
error[E0080]: evaluation of constant value failed
   --> src/lib.rs:285:38
    |
285 |         const _ASSERT_NON_ZERO: [(); $n - 1] = [(); $n - 1];
    |                                      ^^^^^^ attempt to compute `0_usize - 1_usize`, which would overflow
...
363 |         let _ = ne_vec![1; 0];
    |                 ------------- in this macro invocation
    |
    = note: this error originates in the macro `ne_vec` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
Oh, I completely forgot that it's possible to create a macro rule with a hardcoded $n: `($elem:expr; 0) => { ::std::compile_error!("...") }`